packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -622,4 +628,44 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
 
         socket.removeSocketIoReference();
     }
+
+    /**
+     * Key: clientSequenceNumber
+     * Value: content
+     */
+    private readonly opContent = new Map<number, any>();
+
+    public submit(messages: IDocumentMessage[]): void {
+        if ((this.details as any).supportedFeatures?.[feature_light_op_acks] === true) {
+            messages.forEach((message) => this.opContent.set(message.clientSequenceNumber, message.contents));
+        }
+        super.submit(messages);
+    }
+
+    public processInboundMessages(messages: ISequencedDocumentMessage[]): ISequencedDocumentMessage[] {
+        if ((this.details as any).supportedFeatures?.[feature_light_op_acks] !== true) {
+            return messages;
+        }
+        return messages.map((message) => {
+            if (this.clientId === message.clientId && this.opContent.has(message.clientSequenceNumber)) {
+                message.contents = this.opContent.get(message.clientSequenceNumber);
+                this.opContent.delete(message.clientSequenceNumber);
+            }
+            return message;
+        });
+    }
+
+    protected emitMessages(type: string, messages: IDocumentMessage[][]) {
+        if ((this.details as any).supportedFeatures?.[feature_light_op_acks] !== true) {
+            return super.emitMessages(type, messages);
+        }
+        if (!this.disposed) {
+            this.socket.emit(
+                type,
+                this.clientId,
+                true, // support light op acks
+                messages,
```

Comment:
I'm a bit at a loss here re flow.
I believe calling into super, and emitting ops directly will result in same ops being emitted on the socket twice, right?
I'm also not sure why behavior should be different whether new feature is supported by service or not. The only difference on sending path is - should we remember extra content for future or not.